projects/core/src/window/window-ref.ts

```
@@ -58,6 +59,21 @@ export class WindowRef {
     return this.nativeWindow ? this.nativeWindow.localStorage : undefined;
   }
 
+  /**
+   * Returns the window/document location, unless it's not available in the process (i.e. SSR).
+   * When there's no access to the location object, we mimic the location partially, by resolving
+   * the request url (`serverRequestUrl`) and origin (`serverRequestOrigin`) from the injector.
+   *
+   */
+  get location(): Partial<Location> {
+    return !this.isBrowser()
+      ? this.document.location
+      : {
+          href: this.serverUrl || '',
+          origin: this.serverOrigin || '',
```

[nitpick] If those values are undefined/null in SSR, then it means something is seriously misconfigured or not working. IMO we should not silently fallnback to `''` in SSR, but rather leave it undefined/null or `throw new Error(...)`

---

agreed